### DISCUSSION TITLE
first and last spec. before/after cypress hooks

### DISCUSSION THREAD

**Discussion by alexneo2003** [sorry-cypress/sorry-cypress#387] [Q&A]
i'm have 34 spec files 
in support file defined 'before' hook with precondition tasks which requests API for destroy DB and init some startup data for tests
i'm need 'before' hook to be called only ONCE - before first spec will be running

in support file defined 'after' hook with tasks which requests API for destroy DB 
i'm need 'after' hook to be called only ONCE - when last spec will be running

to solve this problem i'm using custom solution which load all specs from folder and use forEach to run tests with cypress module and pass each 'spec' file as cypress args. 
parallel to this script detect index of spec and for 'first' spec it passes the value to cypress env 'setupSuite=true', for 'last' spec 'teardownSuite=true'
and when cypress execute before/after hooks i'm get setupSuite/teardownSuite from env and call API based on value setupSuite/teardownSuite

any ideas to solve this case with sorry-cypress?)

**Comment by agoldis** [comment]
You mean running before / after hooks for the whole run instead of individual tests?

**Reply by alexneo2003** [reply]
Yes, that's it what i mean

**Reply by agoldis** [reply]
hm... I don't think we can do that. In theory, the tests should be independent - i.e. you can run those individually and that includes setup / tear down.

**Reply by alexneo2003** [reply]
Ok, I'm understood. Thanks
